validation/src/main/scala/hmda/validation/api/ValidationApi.scala

```diff
@@ -0,0 +1,24 @@
+package hmda.validation.api
+
+import hmda.validation.dsl.{ Failure, Result, Success }
+import hmda.validation.engine.ValidationError
+import scalaz._
+import scalaz.Scalaz._
+
+trait ValidationApi {
+
+  def convertResult[A](input: A, result: Result, ruleName: String): ValidationNel[ValidationError, A] = {
```

Comment:
Given that `T` is a type parameter, it can be "anything". So I don't expect a particular list of specific types to go here. The `convertResult`method was "less generic" initially but was refactored as part of [this](https://github.com/cfpb/hmda-platform/pull/89/commits/434ba98c032ef73c293b11c6f91d3871482f7d0a) commit to reduce boilerplate. 

Besides `TransmittalSheet` and `LoanApplicationRecord` there has been some talk recently of using a similar approach for validating Panel construction. So you can think of another set of "rules" for that process, in that case this could become very useful.